File: 06ros/ws02/first_ws/src/use_custom4_msg_pkg/scripts/publisher_cus4_msg.py
#! /usr/bin/env python
# coding:utf8

import rospy
from custom4_msgs.msg import Team
from custom_msgs.msg import Student

if __name__ == '__main__':

    nodeName = "publisher_node"
    rospy.init_node(nodeName)

    topicName = "/zcb01/topic"
    pub = rospy.Publisher(topicName, Team, queue_size=1000)

    rate = rospy.Rate(2)
    msg = Team()
    leader = Student()

    mem1 = Student()
    mem2 = Student()
    mem3 = Student()

    idx = 0
    while not rospy.is_shutdown():
        msg.name = "NCHU%d" % idx
        leader.name = "tom%d" % idx
        leader.age = idx
        msg.leader = leader
        msg.intro.data = "NIU bi 战队"

        msg.members = []
        mem1.name = "成员01";mem1.age = idx;msg.members.append(mem1)
        mem2.name = "成员02";mem2.age = idx + 1;msg.members.append(mem2)
        mem3.name = "成员03";mem3.age = idx + 2;msg.members.append(mem3)

        pub.publish(msg)

        idx += 1
        rate.sleep()

File: 06ros/ws02/first_ws/src/use_custom4_msg_pkg/src/publisher_cus4_msg.cpp
//
// Created by zcb on 5/3/21.
//

#include <ros/ros.h>
#include <iostream>
#include <custom4_msgs/Team.h>


int main(int argc, char **argv) {

    std::string nodeName = "publisher_node";
    ros::init(argc, argv, nodeName);
    ros::NodeHandle node;

    std::string topicName = "/zcb01/topic";
    ros::Publisher publisher = node.advertise<custom4_msgs::Team>(topicName, 1000);

    custom4_msgs::Team msg;
    custom_msgs::Student leader;

    custom_msgs::Student mem1;
    custom_msgs::Student mem2;
    custom_msgs::Student mem3;

    ros::Rate rate(0.5);
    int idx = 0;
    while (ros::ok()) {
        msg.name = "NCHU" + std::to_string(idx);
        leader.name = "tom" + std::to_string(idx);
        leader.age = idx;
        msg.leader = leader;
        msg.intro.data = "niu bi 战队";

        msg.members.clear();
        mem1.name = "成员01";mem1.age = idx;msg.members.push_back(mem1);
        mem2.name = "成员02";mem2.age = idx+1;msg.members.push_back(mem2);
        mem3.name = "成员03";mem3.age = idx+2;msg.members.push_back(mem3);

        publisher.publish(msg);

        idx++;
        rate.sleep();
    }

    return 0;
}